### PR TITLE
Build step by step pages

### PR DESCRIPTION
**Pull request by deborahchua**
https://trello.com/c/I7BkEdQz/522-build-pages-for-step-by-step-in-collections-publisher

### CODE REVIEW

File: app/views/step_by_step_pages/index.html.erb [hunk 1]
```diff
@@ -1,2 +1,59 @@
...
+    <% @step_by_step_pages.each do |step_by_step_page| %>
+      <tr>
+        <td>
+          <%= step_by_step_page.title %>
+        </td>
+        <td>
+          <%= link_to 'Edit content', step_by_step_page %>
+        </td>
+        <td>
+          Edit rules
```
Comment (selfthinker) [inline app/views/step_by_step_pages/index.html.erb]: I don't think we should add anything that isn't functional yet. I would simply remove any references to the rules.

File: app/views/step_by_step_pages/index.html.erb [hunk 2]
```diff
@@ -1,2 +1,59 @@
...
+        </td>
+        <td>
+          <%= link_to("Preview", step_by_step_page_preview_path(step_by_step_page)) %>
+        </td>
+        <td>
+          Download JSON
```
Comment (selfthinker) [inline app/views/step_by_step_pages/index.html.erb]: Same here. I would simply remove this as it isn't implemented yet.

File: app/views/step_by_step_pages/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,59 @@
...
+        </td>
+        <td>
+          <%= link_to 'Delete', step_by_step_page, method: :delete, data: { confirm: 'Are you sure?' } %>
+        </td>
+        <td>
+          <span class="label label-success">Status</span>
```
Comment (selfthinker) [inline app/views/step_by_step_pages/index.html.erb]: Same here.

File: app/views/step_by_step_pages/index.html.erb [hunk 4]
```diff
@@ -1,2 +1,59 @@
...
+        </td>
+      <% if false %>
```
Comment (selfthinker) [inline app/views/step_by_step_pages/index.html.erb]: I guess this is the equivalent of a comment, which I would probably remove until needed.

File: app/views/step_by_step_pages/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+<%= @step_by_step_page.introduction %>
+
+<p>Meta description: <%= @step_by_step_page.description %></p>
+
+<p class="text-right">
+  <%= link_to("Preview", step_by_step_page_preview_path(@step_by_step_page), class: "btn btn-info") %>
+  <%= link_to 'Edit title, intro or slug', edit_step_by_step_page_path(@step_by_step_page), class: "btn btn-primary" %>
+  <%= link_to 'Add a new step', new_step_by_step_page_step_path(@step_by_step_page), class: "btn btn-primary" %>
+  [Reorder steps]
```
Comment (selfthinker) [inline app/views/step_by_step_pages/show.html.erb]: Same here.

File: app/views/step_by_step_pages/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+<%= render 'partials/form_notice', message: notice, status: "success" %>
+
+<h1><%= @step_by_step_page.title %></h1>
+
+<%= @step_by_step_page.introduction %>
+
+<p>Meta description: <%= @step_by_step_page.description %></p>
+
+<p class="text-right">
+  <%= link_to("Preview", step_by_step_page_preview_path(@step_by_step_page), class: "btn btn-info") %>
```
Comment (selfthinker) [inline app/views/step_by_step_pages/show.html.erb]: I wonder if we should even remove the preview? The current implementation is only very basic and has non-functional elements as well (mainly the "share preview link" link).

**Comment by selfthinker** [comment]
I've just added the content changes Jon and me did yesterday.

I still cannot save the base path (now renamed "slug" in the UI), I always get an error that it's invalid. I tried several variations (starting or ending with slash or not). Is there any specific rule I need to follow? If yes, that should be made explicit in the UI.

**Comment by selfthinker** [comment]
I have removed every reference to anything which wasn't implemented yet or wasn't implemented properly (like the preview). I couldn't remove ae648e17921f764dce0d6a8f26cab6851097cc36 from the git history because that contains other things than just adding the preview page, despite its commit message.

**Comment by selfthinker** [comment]
I noticed that adding or editing a nav and editing a step all show a success feedback message, but adding a new step doesn't. Is that intentional?

**Comment by selfthinker** [comment]
I've added a feedback message when a new step gets created (in the "add step pages" commit).

**Comment by sihugh** [comment]
I've fixed the tests and squashed a few of the last commits

**Comment by deborahchua** [comment]
Looks good @sihugh , thanks for fixing those tests!

**Comment by selfthinker** [comment]
I've just added the latest design changes as well and cleaned up the git history so everything that belongs together is in one commit.

**Comment by selfthinker** [comment]
I've also just moved the 'partials' from the partials folder into the existing 'shared' folder into a new 'steps' subfolder. I think it makes more sense as that is the place where there already are a few shared partials.

I'm happy with all the frontend stuff.

**Pull request merged**